crates/red_knot_python_semantic/resources/mdtest/binary/classes.md

```diff
@@ -0,0 +1,17 @@
+# Binary operations on classes
+
+## Union of two classes
+
+Unioning two classes via the `|` operator is only available in Python 3.10 and later.
+
+```toml
+[environment]
+python-version = "3.10"
+```
+
+```py
+class A: ...
+class B: ...
+
+reveal_type(A | B)  # revealed: UnionType
```

Comment:
This comes from the typeshed.  We might consider special-casing the `BitOr` of two `ClassLiteral`s to return a more specific `Type::Union`

Reply:
(Note that this is not a type annotation, so none of the logic in `Type::in_type_expression` applies)

Reply:
I went ahead and added this special case, but retaining the behavior that it's only available in Python ≥3.10

Reply:
Oh, I commented on this in my review; I think the added special case is not correct, the type from typeshed is :)